wakuv2/waku.go

```
@@ -1012,11 +1012,22 @@ func (w *Waku) publishEnvelope(envelope *protocol.Envelope, publishFn publishFn,
 	defer w.wg.Done()
 
 	var event common.EventType
-	if err := publishFn(envelope, logger); err != nil {
-		logger.Error("could not send message", zap.Error(err))
-		event = common.EventEnvelopeExpired
-	} else {
-		event = common.EventEnvelopeSent
+	for {
+		err := publishFn(envelope, logger)
+		if err != nil {
+			if err.Error() == "not enough peers to publish" {
```

I think the correct approach here would be to create in go-waku:
```
var ErrNotEnoughPeersToPublish = errors.New("not enough peers to publish"
```
and then, compare using  `errors.Is(err, relay.ErrNotEnoughPeersToPublish)`.
That way we ensure that if the error string changes in go-waku, status-go would not be affected by this and fail silently